From ac2ae4ac22d6acef244f900d4b7848607efadc1d Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 26 Oct 2020 17:50:06 +0800
Subject: [PATCH 38/39] HACK: backend-drm: Workaround atomic flip for BSP 4.4
 kernel

The Rockchip BSP 4.4 kernel might generate unexpected flip events
when dpms changed.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/backend-drm/kms.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libweston/backend-drm/kms.c b/libweston/backend-drm/kms.c
index 94cb2a3..c2fce5a 100644
--- a/libweston/backend-drm/kms.c
+++ b/libweston/backend-drm/kms.c
@@ -1525,7 +1525,11 @@ atomic_flip_handler(int fd, unsigned int frame, unsigned int sec,
 
 	drm_debug(b, "[atomic][CRTC:%u] flip processing started\n", crtc_id);
 	assert(b->atomic_modeset);
-	assert(output->atomic_complete_pending);
+
+	/* HACK: Disable this check sink the Rockchip BSP 4.4 kernel might
+	 * generate unexpected flip events when dpms changed. */
+	//assert(output->atomic_complete_pending);
+
 	output->atomic_complete_pending = false;
 
 	drm_output_update_complete(output, flags, sec, usec);
-- 
2.20.1

